tests/post/steps/test_service_configuration.py

```diff
@@ -0,0 +1,271 @@
+    path,
+    value
+):
+    csc_response = read_namespace_config_map(k8s_client, name, namespace)
+
+    assert csc_response, (
+        "No ConfigMap with name {} in namespace {} found".format(
+            name, namespace
+        )
+    )
+
+    csc_obj = yaml_load(csc_response, name, namespace)
+    response_value = traversed_dict(csc_obj, path)
+
+    assert int(value) == response_value, (
+        "Expected value {} for key {} in ConfigMap {} found in namespace {}, "
+        "got {}".format(value, path, name, namespace, response_value)
+    )
+
+
+# }}}
+
+
+# When {{{
+
+
+@when(parsers.parse(
+    "we update '{name}' CSC in namespace '{namespace}' "
+    "'{path}' to '{value}'"))
+def update_service_configuration(
+    host,
+    k8s_client,
+    name,
+    namespace,
+    path,
+    value
+):
+
+    full_csc = read_namespace_config_map(k8s_client, name, namespace)
+    csc_obj = yaml_load(full_csc, name, namespace)
+
+    csc_obj['spec']['deployment']['replicas'] = int(value)
+    patch = \
+        {'data': {'config.yaml': yaml.safe_dump(
+            csc_obj, default_flow_style=False)
+        }}
+
+    response = patch_namespace_config_map(
+        k8s_client, name, namespace, patch
+    )
+
+    assert response
+
+    patched_csc = read_namespace_config_map(
+        k8s_client, name, namespace
+    )
+    config_obj = yaml_load(patched_csc, name, namespace)
+    patched_value = traversed_dict(config_obj, path)
+
+    assert int(value) == patched_value, (
+        "Expected value {} for key {} in ConfigMap {} found in namespace {}, "
+        "got {}".format(value, path, name, namespace, patched_value)
+    )
+
+
+@when(parsers.parse("we apply the '{state}' salt state"))
+def apply_service_config(host, version, request, k8s_client, state):
+    ssh_config = request.config.getoption('--ssh-config')
+
+    cmd = [
+        'salt-run', 'state.sls', '{}'.format(state),
+        'saltenv=metalk8s-{}'.format(version)
+    ]
+    output = utils.run_salt_command(host, cmd, ssh_config)
+
+    assert output.exit_status == 0, \
+        'Failed applying salt-state {}: \nout: {}\nerr:'.format(
+            state,
+            output.stdout,
+            output.stderr
+        )
+
+
+# }}}
+
+
+# Then {{{
+
+
+@then(parsers.parse(
+    "we have '{value}' at '{path}' for '{service}' Deployment in namespace "
+    "'{namespace}'"))
+def get_deployments(
+    host,
+    version,
+    k8s_appsv1_client,
+    k8s_client,
+    request,
+    value,
+    path,
+    service,
+    namespace
+
+):
+    def _wait_for_deployment():
+        try:
+            response = k8s_appsv1_client.read_namespaced_deployment(
+                name=service, namespace=namespace
+            )
+        except Exception as exc:
+            pytest.fail(
+                "Unable to read {} Deployment with error: {!s}".format(
+                    service, exc
+                )
+            )
+
+        assert int(value) == eval("response"+"{}".format(path)), (
```

Comment:
??? I think you forgot to use `traversed_dict` here.

And global python comment: better to avoid using `eval` as much as possible

Reply:
Yeah, `eval` is a no-go.